base/uk.ac.stfc.isis.ibex.ui.journalviewer/src/uk/ac/stfc/isis/ibex/ui/journalviewer/SearchInput.java

```diff
@@ -0,0 +1,309 @@
+        lblSpacing1.setText("        ");
+
+        chkTimeTo = new Button(cmpTimePicker, SWT.CHECK);
+        chkTimeTo.setText("To");
+
+        dtToDate = new DateTime(cmpTimePicker, SWT.BORDER);
+        dtToDate.setEnabled(false);
+
+        dtToTime = new DateTime(cmpTimePicker, SWT.BORDER | SWT.TIME);
+        dtToTime.setEnabled(false);
+
+        chkTimeFrom.addSelectionListener(new SelectionAdapter() {
+            @Override
+            public void widgetSelected(SelectionEvent e) {
+                dtFromDate.setEnabled(chkTimeFrom.getSelection());
+                dtFromTime.setEnabled(chkTimeFrom.getSelection());
+            }
+        });
+
+        chkTimeTo.addSelectionListener(new SelectionAdapter() {
+            @Override
+            public void widgetSelected(SelectionEvent e) {
+                dtToDate.setEnabled(chkTimeTo.getSelection());
+                dtToTime.setEnabled(chkTimeTo.getSelection());
+            }
+        });
+
+        cmpRBNumber = new Composite(cmpSearch, SWT.NONE);
+        cmpFilters.add(cmpRBNumber);
+        cmpRBNumber.setLayout(new GridLayout(1, false));
+
+        txtSearchRBNumber = new Text(cmpRBNumber, SWT.BORDER);
+        GridData gdTxtSearchRBNumber = new GridData(SWT.LEFT, SWT.CENTER, false, false, 1, 1);
+        gdTxtSearchRBNumber.widthHint = SEARCH_BOX_WIDTH;
+        txtSearchRBNumber.setLayoutData(gdTxtSearchRBNumber);
+
+        stackSearch.topControl = cmpRunNumber;
+
+        cmpUsers = new Composite(cmpSearch, SWT.NONE);
+        cmpFilters.add(cmpUsers);
+        cmpUsers.setLayout(new GridLayout(1, false));
+
+        txtSearchUsers = new Text(cmpUsers, SWT.BORDER);
+        GridData gdTxtSearchUsers = new GridData(SWT.LEFT, SWT.CENTER, false, false, 1, 1);
+        gdTxtSearchUsers.widthHint = SEARCH_BOX_WIDTH;
+        txtSearchUsers.setLayoutData(gdTxtSearchUsers);
+
+        cmpSearch.setTabList(new Control[] {cmpRunNumber, cmpTitle, cmpTimePicker, cmpRBNumber, cmpUsers});
+
+        chkNumberFrom.addSelectionListener(new SelectionAdapter() {
+            @Override
+            public void widgetSelected(SelectionEvent e) {
+                spinnerFromNumber.setEnabled(chkNumberFrom.getSelection());
+            }
+        });
+
+        chkNumberTo.addSelectionListener(new SelectionAdapter() {
+            @Override
+            public void widgetSelected(SelectionEvent e) {
+                spinnerToNumber.setEnabled(chkNumberTo.getSelection());
+            }
+        });
+
+        cmbFilterType.addSelectionListener(new SelectionAdapter() {
+            @Override
+            public void widgetSelected(SelectionEvent e) {
+                DISPLAY.asyncExec(new Runnable() {
+                    @Override
+                    public void run() {
+                        stackSearch.topControl = cmpFilters.get(cmbFilterType.getSelectionIndex());
+                        cmpSearch.layout();
+                    }
+                });
+            }
+        });
```

Comment:
This method is very long. UI methods are allowed to be a bit longer than logic methods, but this method is still a bit excessive. Is there a way to split it up?

Reply:
I've moved the listeners and bindings out of the method and got rid of some duplication, which shortened it from 145 lines to 90 lines.